portfolio/src/main/java/com/google/sps/servlets/DoggoRedditServlet.java

```diff
@@ -0,0 +1,127 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+import java.io.PrintWriter;
+
+@WebServlet("/doggo-servlet")
+public final class DoggoRedditServlet extends HttpServlet {
+
+  static final String COMMENT_TEXT = "commentText";
+  static final String TIMESTAMP = "timestamp";
+  static final String IMAGE_URL = "imageUrl";
+  static final String FEED_POST = "FeedPost";
+  static final String DOG_NAME = "dogName";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Entity postEntity = new Entity(FEED_POST);
+    long timestamp = System.currentTimeMillis();
+    String uploadURL = getUploadedFileUrlFromBlobstore(request, "image");
+
+    postEntity.setProperty(DOG_NAME, request.getParameter(DOG_NAME));
+    postEntity.setProperty(COMMENT_TEXT, request.getParameter(COMMENT_TEXT));
+    postEntity.setProperty(TIMESTAMP, timestamp);
+    postEntity.setProperty(IMAGE_URL, uploadURL);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(postEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Query query = new Query(FEED_POST).addSort(TIMESTAMP, SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<FeedPost> posts = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      
+      String dogName = (String) entity.getProperty(DOG_NAME);
+      String commentText = (String) entity.getProperty(COMMENT_TEXT);
+      String imageUrl = (String) entity.getProperty(IMAGE_URL);
+      long timestamp = (long) entity.getProperty(TIMESTAMP);
+      long id = entity.getKey().getId();
+
+      FeedPost post = new FeedPost(dogName, commentText, imageUrl, timestamp);
+      posts.add(post);
+    }
+
+    Gson gson = new Gson();
```

Comment:
make this a class member variable so you don't have to reconstruct it every time

Reply:
Done!